sycl/plugins/hip/pi_hip.cpp

```diff
@@ -1655,7 +1655,7 @@ pi_result hip_piDeviceGetInfo(pi_device device, pi_device_info param_name,
     // DEVICELIB_ASSERT extension is set so fallback assert
     // postprocessing is NOP. HIP 4.3 docs indicate support for
     // native asserts are in progress
-    std::string SupportedExtensions = "";
+    std::string SupportedExtensions = "cl_khr_fp64 ";
```

Comment:
Could you add a test to llvm-test-suite or if you can make it compiler only add it to the compiler only tests in this repo? This would stop the issue from coming back.

Reply:
Do all existing HIP devices support `cl_khr_fp64` functionality?

Reply:
I have done some digging, I cannot find any indication that different devices have different precision support. So, I would assume all HIP devices support fp64

Reply:
So from looking into HIP further it seems like you can query from the host if the device supports `fp64`. 
https://github.com/ROCm-Developer-Tools/HIP/blob/0acac9c7db45c8e3c107198eebfa83cb031715c4/include/hip/hip_runtime_api.h#L58

You can use `hipGetDeviceProperties` to get the `hipDeviceProp_t` and in this struct there is the `hipDeviceArch_t`.

So perhaps this is what we should do rather than assume the device supports fp64. Also now as it is clear the device cannot be guaranteed to have fp64 then I don't see a way of writing a valid test, so it is fine that we don't add one along with this PR.